tbtadm/controller.cpp

```diff
@@ -336,12 +336,12 @@ void tbtadm::Controller::devices()
             continue;
         }
 
-        chdir(dir.path());
```

Comment:
When we start using polkit, I think we'll need changing the dir and just use the current dir for `tbtadm-hepler` to know what device t approve, as otherwise an attacker can trick the user to send "approve" for a device and replace it before the `tbtadm-helper` comes to actually approve it.
What do you think?

Reply:
I am not sure using chdir() would help with it.

Reply:
It is.
I tested it (with `tbtacl`) by adding a sleep and changing devices between the ACL check and the authorization and the authorization failed, as the current dir has gone from the filesystem (even if there is now a new one with the same name).
The same goes when using the sysfs interface directly from shell. If you `cd` to the device directory and then replace it (or disconnect and reconnect it), `echo 1 > authorized` or even simple `ls` will fail and you have to do `cd $PWD` first.

Reply:
I do not get how it could be expoited. As analogue to shell going to the directory before writing 1 does not look secure enough ;)

Reply:
(Just to be clear: for this PR it's fine. It's just if we start using polkit that we'll have to rethink about it.)